java/src/shared/facades/DevCardFacade.java

```diff
@@ -88,7 +88,11 @@ public void buyDevCard(@NotNull Player currentPlayer) {
             bank.setOfType(t, bank.getOfType(t) - all.getOfType(t));
         }
         DevCardType randomType = bank.getRandom();
-        currentPlayer.getNewDevCards().setOfType(randomType, currentPlayer.getNewDevCards().getOfType(randomType) + 1);
+        if (randomType == DevCardType.MONUMENT) {
```

Comment:
Does the demo server put monument cards in the old development cards? IIRC the controller should handle Monument cards specially to work even if they're in `newDevCards`.